src/app/config/config.c

```diff
@@ -1481,15 +1481,109 @@ options_act_once_on_startup(char **msg_out)
   return 0;
 }
 
+/**
+ * Change our user ID if we're configured to do so.
+ **/
+static int
+options_switch_id(char **msg_out)
```

Comment:
We already have a switch_id test, we should run it as root to check this refactor.

Reply:
Good idea -- the test passes.